libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1216,8 +1236,10 @@ bool AC_PosControl::calculate_yaw_and_rate_yaw()
     if (vel_desired_xy_len > _vel_max_xy_cms * 0.05f) {
         _yaw_target = degrees(_vel_desired.xy().angle()) * 100.0f;
         _yaw_rate_target = turn_rate * degrees(100.0f);
+        _yaw_target_valid = true;
```

Comment:
This function already returns true/false. You could set `_yaw_target_valid` here:

https://github.com/ArduPilot/ardupilot/blob/84b3c9e2674aabb7f9d91e058d1d77c2e5af8adc/libraries/AC_AttitudeControl/AC_PosControl.cpp#L656

eg:
```
    _yaw_target_valid = calculate_yaw_and_rate_yaw();
```

Or I guess turn this into a void function, there is only one caller which is not using the return.

Reply:
Thanks I've made it a void function because, as you say, noone was using the return.  Also I think it is best to set the _yaw_target_valid flag at the same time as we set _yaw_target.